packages/hadron-build/commands/upload.js

```diff
@@ -1,139 +1,458 @@
+}
+
+async function uploadAssetsToDownloadCenter(assets, channel, dryRun) {
+  const assetsToUpload = assets
+    .flatMap((item) => {
+      return item.assets;
+    })
+    .filter(({ downloadCenter }) => {
+      return downloadCenter;
+    });
+
+  cli.info('Uploading assets to download center…');
+
+  await checkAssetsExist(
+    assetsToUpload.map((asset) => {
+      return asset.path;
+    })
   );
 
+  const uploads = assetsToUpload.map(async(asset) => {
+    cli.info(
+      `${asset.name}: upload to download center started (path: ${path.relative(
+        root,
+        asset.path
+      )}).`
+    );
+    if (!dryRun) {
+      await uploadAsset(channel, asset);
+    }
+    cli.info(`${asset.name}: upload to download center completed.`);
+  });
+
   await Promise.all(uploads);
 }
 
-exports.command = 'upload [options]';
+/**
+ *
+ * @param {'stable' | 'beta'} channel
+ */
+async function getLatestRelease(channel = 'stable') {
+  const octokit = new Octokit({
+    auth: process.env.GITHUB_TOKEN
+  });
+
+  const page = 1;
 
-exports.describe = 'Upload assets from `release`.';
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    let releases = [];
 
-exports.builder = {
+    try {
+      const { data } = await octokit.request(
+        'GET /repos/{owner}/{repo}/releases',
+        {
+          owner: 'mongodb-js',
+          repo: 'compass',
+          per_page: 100,
+          page
+        }
+      );
+      releases = data;
+    } catch (err) {
+      cli.warn(`Failed to fetch releases: ${err.message}`);
+    }
+
+    // We ran out of releases or failed to fetch
+    if (releases.length === 0) {
+      return null;
+    }
+
+    const latestRelease = releases.find((release) => {
+      return (
+        !release.draft &&
+        (channel === 'beta'
+          ? isBeta(release.tag_name)
+          : isStable(release.tag_name))
+      );
+    });
+
+    if (latestRelease) {
+      return latestRelease;
+    }
+
+    page++;
+  }
+}
+
+async function getLatestReleaseVersions(channel = 'stable') {
+  const release = await getLatestRelease(channel);
+  if (!release) {
+    throw new Error(`Couldn't find latest release for ${channel} channel`);
+  }
+  const manifest = release.assets.find((asset) => {
+    return asset.name === 'manifest.json';
+  });
+  if (!manifest) {
+    throw new Error(`No manifest found in the release for ${channel} channel`);
+  }
+  const content = JSON.parse(
+    (await download(manifest.browser_download_url)).toString()
+  );
+  return content;
+}
+
+async function updateManifest(dryRun) {
+  cli.info('Downloading current manifest');
+  const currentManifest = await downloadManifest();
+
+  const versions = (
+    await Promise.all(
+      ['stable', 'beta'].map((channel) => {
+        cli.info(`Looking for the latest release for the ${channel} channel`);
+        return getLatestReleaseVersions(channel);
+      })
+    )
+  ).flat();
+
+  const newManifest = {
+    ...currentManifest,
```

Comment:
I really think we shouldn't do this and rather generate a fresh new one ignoring anything that was there before. Any particular reason why we want to rather modify the old one?

Reply:
I don't really have a preference so will change. There is only a handful of metadata fields here so it doesn't seem that important if we use the old ones or just have them hardcoded in this file

Reply:
Updated in 577bb263683a7bccad62f5cda0656ffc39262f53

Reply:
Thanks! I like it a lot better to have the entire file here and avoid any mutation, like that we can control everything in this place and if we would ever need to change it we can do it with a PR

Reply:
Yep, this sounds like a good reason!